src/graph/backend/elyzor/elyzor_partition_impl.hpp

```diff
@@ -27,13 +27,16 @@
 #include "elyzor_backend.hpp"
 #include "utils.hpp"
 
+class test_elyzor_partition_impl_t;
+
 namespace dnnl {
 namespace impl {
 namespace graph {
 namespace elyzor {
 
 class elyzor_partition_impl_t : public partition_impl_t {
     friend class elyzor_backend_t;
+    friend class ::test_elyzor_partition_impl_t;
```

Comment:
need this to access protected `copied_ops_` field

Reply:
Is this for testing only?

Reply:
yep

Reply:
May be add a public getter and wrap it into `#ifndef NDEBUG`?

Reply:
> wrap it into #ifndef NDEBUG?

Probably not `NDEBUG` but rather `#ifdef UNIT_TESTING` as tests are not necessarily built in debug mode. 

I don't really like this approach, as in the case of oneDNN, the tests are [built by default](https://oneapi-src.github.io/oneDNN/dev_guide_build_options.html#:~:text=building%20the%20examples-,ONEDNN_BUILD_TESTS,-ON%20%2C%20OFF), meaning that `#ifdef UNIT_TESTING` condition is always true and the getter function we wanted to hide is accessible to everyone.

In the case of a friend class, only the testing class can access private attributes of the partition and nobody else

Reply:
You could also use NDEBUG in the tests and disable the cases that use that getter. I afraid the current approach may not be accepted when merging upstream.
Another option how to get access to the protected field - add a copy constructor to elyzor_partition_impl_t and subclass it in the test.